posthog/migrations/0306_add_indirect_person_override_constraints.py

```diff
@@ -0,0 +1,96 @@
+# Generated by Django 3.2.16 on 2023-03-01 17:03
+
+import django.contrib.postgres.constraints
+import django.db.models.deletion
+import django.db.models.expressions
+from django.db import migrations, models
+
+# The previous migration attempted to add a constraint to the personoverride
+# table. We want to remove that constraint, as the ForeignKey replaces this.
+# The function was never used, but to make migrations work we need these two
+DROP_FUNCTION_FOR_CONSTRAINT_SQL = "DROP FUNCTION is_override_person_not_used_as_old_person"
+CREATE_FUNCTION_FOR_CONSTRAINT_SQL = f"""
+CREATE OR REPLACE FUNCTION is_override_person_not_used_as_old_person(team_id bigint, override_person_id uuid, old_person_id uuid)
+RETURNS BOOLEAN AS $$
+  SELECT false;
+$$ LANGUAGE SQL;
+"""
+
+DROP_FUNCTION_FOR_CONSTRAINT_SQL = "DROP FUNCTION is_override_person_not_used_as_old_person"
+
+
+class Migration(migrations.Migration):
```

Comment:
This is a thick migration. But unsafe operations are limited to models that are not in use currently (`PersonOverrides` mostly).

Notice we are not dropping the `team` FK from `PersonOverrides` even though it's redundant. This is to avoid table locks on `Team` during migration. `Team`, in contrast to `PersonOverrides` is used a lot so locking is bad.